### PR TITLE
[DropZone] Change DOM order of Visually hidden input in order to focus it before its children

### PR DESCRIPTION
**Pull request by husshegz**
### WHY are these changes introduced?

We have a use-case where there is a `Button` inside the `dropzone`

Currently, if we're navigating (tabbing) through the component with the keyboard:
1. `Button` inside `Dropzone` is focused first
2. `Dropzone` is focused

We expect a meaningful order when navigating through. Expected order:
1. Focus the `Dropzone` first
2. Then any focusable contents inside it

<details>

<summary> Current </summary>

https://user-images.githubusercontent.com/23648577/119017743-4e391300-b969-11eb-9c15-21f6fffaca9b.mp4

</details>

<details>

<summary> Correct order </summary>

https://user-images.githubusercontent.com/23648577/119017814-63ae3d00-b969-11eb-8bfe-f91a9b9d1c8f.mp4

</details>

### WHAT is this pull request doing?

Changes DOM order to accommodate a meaningful tabbing sequence. 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

Could this possibly break anything from the consumer side ? 

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by husshegz** [comment]
@jared-fanning 
Let me know if this needs more changes, tests, story edit. I couldn't come up with any relevant examples. 

Also should I update the `UNRELEASED` before or after approval ?

**Comment by kyledurand** [comment]
> Also should I update the UNRELEASED before or after approval ?

I think it makes sense to add a note for this. I'll approve when it's been added just ping me 👍

**Review by kyledurand — APPROVED** [review]
👍 Thanks!

**Pull request merged**
